packages/agent/src/polling/index.ts

```diff
@@ -29,8 +29,9 @@ export async function pollForResponse(
 ): Promise<ArrayBuffer> {
   const path = [new TextEncoder().encode('request_status'), requestId];
   const state = await agent.readState(canisterId, { paths: [path] });
-  const cert = new Certificate(state, agent);
-  const verified = await cert.verify();
+  const rootKey = agent.rootKey == null ? agent.fetchRootKey() : Promise.resolve(agent.rootKey);
+  const cert = new Certificate(state.certificate, rootKey);
+  const verified = await cert.verify(canisterId);
```

Comment:
We don't have sufficient information at this point in the code to compute this (there's no request); per the comment, the function already expects an effective canister ID. I don't see any special computation of the effective ID, but I may have missed something.

So I think you're right and we should add it, but I don't think this PR is breaking anything beyond what's broken already. In particular, to read the status of calls to the management canister other than `provisional_create_canister_with_cycles`, you need to compute the effective canister ID regardless of whether you're properly checking the certificate, and then `read_state` on that ID, otherwise (assuming that the replica actually follows the interface spec) you don't get a response.